In [1]:
import pandas as pd
import statsmodels.api as sm
dataset = pd.read_csv('6.3.3.csv', sep=";") 
dataset

Tlak  Priehyb
0     2       14
1     4       35
2     6       48
3     8       61
4    10       80
5    12       93

In [2]:
import math
dataset['intercept'] = 0
results = sm.OLS(dataset['Priehyb'], dataset[['intercept', 'Tlak']]).fit()
print ("Multiple R = ",math.sqrt(results.rsquared))
results.summary()

Multiple R =  0.9994759367270442


C:\Users\andre\anaconda3\lib\site-packages\statsmodels\stats\stattools.py:74: ValueWarning: omni_normtest is not valid with less than 8 observations; 6 samples were given.
  warn("omni_normtest is not valid with less than 8 observations; %i "
C:\Users\andre\anaconda3\lib\site-packages\statsmodels\regression\linear_model.py:1860: RuntimeWarning: divide by zero encountered in double_scalars
  return np.sqrt(eigvals[0]/eigvals[-1])


<class 'statsmodels.iolib.summary.Summary'>
"""
                                 OLS Regression Results                                
=======================================================================================
Dep. Variable:                Priehyb   R-squared (uncentered):                   0.999
Model:                            OLS   Adj. R-squared (uncentered):              0.999
Method:                 Least Squares   F-statistic:                              4767.
Date:                Sun, 02 Apr 2023   Prob (F-statistic):                    1.21e-08
Time:                        09:43:35   Log-Likelihood:                         -12.618
No. Observations:                   6   AIC:                                      27.24
Df Residuals:                       5   BIC:                                      27.03
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
intercept           0          0        nan        nan           0           0
Tlak           7.8571      0.114     69.041      0.000       7.565       8.150
==============================================================================
Omnibus:                          nan   Durbin-Watson:                   2.581
Prob(Omnibus):                    nan   Jarque-Bera (JB):                0.583
Skew:                           0.515   Prob(JB):                        0.747
Kurtosis:                       1.872   Cond. No.                          inf
==============================================================================

Notes:
[1] R² is computed without centering (uncentered) since the model does not contain a constant.
[2] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[3] The smallest eigenvalue is      0. This might indicate that there are
strong multicollinearity problems or that the design matrix is singular.
"""

Aby priamka prechádzala počiatkom súradnicového systému, museli sme nastaviť absolútny člen na 0. Vo výslednom regresnom modeli má R-štvorcové hodnotu 0.999 a z toho vyplýva, že sme zvolili správny regresný model. Výsledná priamka má tvar: *Priehyb* = 7,8571 * *Tlak*